connmgr/connmanager.go

```diff
@@ -346,6 +352,28 @@ out:
 				log.Debugf("Failed to connect to %v: %v",
 					connReq, msg.err)
 				cm.handleFailedConn(connReq)
+
+			case handleCancelPending:
+				found := false
+				var idToRemove uint64
+				connReq := &ConnReq{}
```

Comment:
This is allocating memory only to replace it a pending request is found or ignore it when it isn't.  It should be nil by default which would also remove the need for the found flag since it would only be non-nil if found.

Also, converting the address to a string multiple times inside the loop is inefficient.  It would suggest defining it outside of the for loop.

Thus, the following would do all of the above:

```Go
pendingAddr := msg.addr.String()
var idToRemove uint64
var connReq *ConnReq
for id, req := range pending {
	if pendingAddr == req.Addr.String() {
		idToRemove, connReq = id, req
		break
	}
}
if connReq != nil {
	...
}
```

Then, you don't need the `found` flag, because it will only be non-nil when it is found.  Ergo: